src/selectors/getCreateWalletList.ts

```diff
@@ -64,82 +64,104 @@ interface CreateWalletListOpts {
 }
 
 export const getCreateWalletList = (account: EdgeAccount, opts: CreateWalletListOpts = {}): WalletCreateItem[] => {
-  const { filteredWalletList = [], filterActivation, allowedAssets, excludeAssets, disableLegacy = false } = opts
+  const { filteredWalletList = [], filterActivation, allowedAssets = [], excludeAssets = [], disableLegacy = false } = opts
+  const segwitSpecialCases = new Set(['bitcoin', 'litecoin', 'digibyte'])
+  const existingWalletsMap = new Map<string, Set<EdgeTokenId>>()
+  for (const item of filteredWalletList) {
+    if (item.type !== 'asset') continue
+    const { wallet, tokenId } = item
+    const tokenIdSet = existingWalletsMap.get(wallet.currencyInfo.pluginId) ?? new Set()
+    tokenIdSet.add(tokenId)
+    existingWalletsMap.set(wallet.currencyInfo.pluginId, tokenIdSet)
+  }
+  const assetOverrides = infoServerData.rollup?.assetOverrides ?? {
+    disable: {}
+  }
+
+  const createAssetMap = (assets: EdgeAsset[]): Map<string, Set<EdgeTokenId>> => {
+    const assetsMap = new Map<string, Set<EdgeTokenId>>()
+    for (const asset of assets) {
+      const tokenIdSet = assetsMap.get(asset.pluginId) ?? new Set()
+      tokenIdSet.add(asset.tokenId)
+      assetsMap.set(asset.pluginId, tokenIdSet)
+    }
+    return assetsMap
+  }
+  const excludedAssetsMap = createAssetMap(excludeAssets)
+  const allowedAssetsMap = createAssetMap(allowedAssets)
+
+  const isAllowed = (pluginId: string, tokenId: EdgeTokenId) =>
+    // if the wallet already exists, then it is not allowed
+    !existingWalletsMap.get(pluginId)?.has(tokenId) &&
+    // if allowedAssets is empty, then all assets are allowed
+    (allowedAssetsMap.size === 0 || allowedAssetsMap.get(pluginId)?.has(tokenId)) &&
+    // if excludedAssets is not empty, then the asset must not be in the excluded list
+    !(excludedAssetsMap.size > 0 && excludedAssetsMap.get(pluginId)?.has(tokenId))
```

Comment:
Applying DeMorgan's Law, we can turn `!(excludedAssetsMap.size > 0 && excludedAssetsMap.get(pluginId)?.has(tokenId))` into `(excludedAssetsMap.size === 0 || !excludedAssetsMap.get(pluginId)?.has(tokenId))`. This second version reads better, and matches the `allowedAssetsMap` logic above, except for the `!` before the `has`.

Reply:
fixed